Modules/Scripted/DICOMLib/DICOMUtils.py

```diff
@@ -156,3 +156,180 @@ def __enter__(self):
+          (pluginName, selectedPlugins[pluginName], actualSelectedPlugins[pluginName]))
+        success = False
+
+  # Count relevant node types in scene
+  actualLoadedNodes = {}
+  for nodeType in loadedNodes.keys():
+    nodeCollection = slicer.mrmlScene.GetNodesByClass(nodeType)
+    nodeCollection.UnRegister(None)
+    actualLoadedNodes[nodeType] = nodeCollection.GetNumberOfItems()
+
+  # Load selected data
+  dicomWidget.detailsPopup.loadCheckedLoadables()
+
+  for nodeType in loadedNodes.keys():
+    nodeCollection = slicer.mrmlScene.GetNodesByClass(nodeType)
+    nodeCollection.UnRegister(None)
+    numOfLoadedNodes = nodeCollection.GetNumberOfItems()-actualLoadedNodes[nodeType]
+    if numOfLoadedNodes != loadedNodes[nodeType]:
+      logging.error("Number of loaded %s nodes was %d, but %d was expected" % \
+        (nodeType, numOfLoadedNodes, loadedNodes[nodeType]) )
+      success = False
+
+  return success
+
+#------------------------------------------------------------------------------
+# Collect all series instance UIDs in a DICOM database (the Slicer one by default)
+# Useful to get list of just imported series UIDs, for example:
+# newSeriesUIDs = [x for x in seriesUIDsAfter if x not in seriesUIDsBefore]
+def allSeriesUIDsInDatabase(database=None):
+  if database is None:
+    database = slicer.dicomDatabase
+  dicomWidget = slicer.modules.dicom.widgetRepresentation().self()
+  allSeriesUIDs = []
+  for patient in database.patients():
+    studies = database.studiesForPatient(patient)
+    series = [database.seriesForStudy(study) for study in studies]
+    seriesUIDs = [uid for uidList in series for uid in uidList]
+    allSeriesUIDs.extend(seriesUIDs)
+  return allSeriesUIDs
+
+#------------------------------------------------------------------------------
+# Collect series instance UIDs belonging to a list of files
+def seriesUIDsForFiles(files):
+  seriesUIDs = set()
+  for file in files:
+    seriesUID = slicer.dicomDatabase.seriesForFile(file)
+    if seriesUID != '':
+      seriesUIDs.add(seriesUID)
+  return seriesUIDs
+
+#------------------------------------------------------------------------------
+class OnlineDICOMData:
+  """Context manager to conveniently load DICOM files downloaded zipped from the internet
+  """
+  def __init__(self, url, zipFilePath=None, dicomDataDir=None, expectedNumOfFiles=0, selectedPlugins={}, loadedNodes={}):
+    from time import gmtime, strftime
+    self.url = url
+    if zipFilePath is None:
+      fileName = strftime("%Y%m%d_%H%M%S_", gmtime()) + 'OnlineDICOMData.zip'
+      self.zipFilePath = slicer.app.temporaryPath + '/' + fileName
+    else:
+      self.zipFilePath = zipFilePath
+    if dicomDataDir is None:
+      directoryName = strftime("%Y%m%d_%H%M%S_", gmtime()) + 'OnlineDICOMData'
+      self.dicomDataDir = slicer.app.temporaryPath + '/' + directoryName
+    else:
+      self.dicomDataDir = dicomDataDir
```

Comment:
I  suggest to set default values first, and update object properties at the end.

```
[...]
+    if zipFilePath is None:
+      fileName = strftime("%Y%m%d_%H%M%S_", gmtime()) + 'OnlineDICOMData.zip'
+      self.zipFilePath = slicer.app.temporaryPath + '/' + fileName
+
+    if dicomDataDir is None:
+      directoryName = strftime("%Y%m%d_%H%M%S_", gmtime()) + 'OnlineDICOMData'
+      dicomDataDir = slicer.app.temporaryPath + '/' + directoryName
+ 
+    self.url = url
+    self.zipFilePath = zipFilePath
+    self.dicomDataDir = dicomDataDir
+    self.expectedNumberOfUnzippedFiles = expectedNumOfFiles
+    self.selectedPlugins = selectedPlugins
+    self.loadedNodes = loadedNodes
```